### PR TITLE
Add versatile image field prewarmer

### LINKED ISSUE #1436: Images generation performance issue
Right now we are using the `django-versatileimagefield` (VF) package for images thumbnails generation. Its default behavior is to generate images on demand. VF uses cache to determine which images need to be generated.

That behavior is dangerous on typical ecommerce products list. If page for instance has 20 images and 4 different sizes per resolution we end up with 80 request to the cache and in worst scenario 80 images start to be generated on machine which is handling responses. That basically kill the site.

In all previous projects we disabled auto generation. Because of that we had to write custom celery tasks (which generate images in the background) and custom template tag (@artursmet can provide more details). It will be nice to have this done by default.

### PR DESCRIPTION
**Pull request by Pacu2**
This pull requests add versatile image field prewarmer for product images.
close #1436 

### Pull Request Checklist

(Please keep this section. It will make maintainer's life easier.)

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

**Comment by NyanKiyoshi** [comment]
It would probably be useful to also create a command for `manage.py` to warm up any (pre)existing image.

**Comment by Pacu2** [comment]
@NyanKiyoshi Thought of the same thing, it will be definitely added.

File: saleor/product/models.py [hunk 1]
```diff
@@ -392,6 +393,15 @@ def delete(self, *args, **kwargs):
         super().delete(*args, **kwargs)
 
 
+@receiver(models.signals.post_delete, sender=ProductImage)
```
Comment (patrys) [inline saleor/product/models.py]: Is there a way not to use signals to implement this?

Comment (Pacu2) [reply saleor/product/models.py]: We should cover all places where ProductImage or its parent can be deleted.
Eg. Category/Product/ProductImage delete actions
Plus something can be always deleted from the shell (edge case,but still, most likely during development phase)

Comment (patrys) [reply saleor/product/models.py]: Do we currently delete the main image? If so, does it also rely on signals? If not, do we want to remove thumbnails?

Comment (elwoodxblues) [reply saleor/product/models.py]: We don't use signals at all, except for one that comes from `django-payments` if I'm correct. Also, we should care about cases when someone does changes using shell.

**Pull request merged**